lib/puppet/provider/package/yum.rb

```diff
@@ -142,13 +142,24 @@ def self.update_command
     'update'
   end
 
+  def makecache_fast
+    command = [command(:cmd), :makecache, :fast]
+    execute(command, :failonfail => false, :combine => false)
+  end
+
   def install
     wanted = @resource[:name]
     error_level = self.class.error_level
     update_command = self.class.update_command
     # If not allowing virtual packages, do a query to ensure a real package exists
     unless @resource.allow_virtual?
-      execute([command(:cmd), '-d', '0', '-e', error_level, '-y', install_options, :list, wanted].compact)
+      command = [command(:cmd), '-d', '0', '-e', error_level, '-y', install_options, :list, wanted].compact
```

Comment:
I'm a bit leery about duplicating this command string. If we need to change the invocation of this command it would be possible to only do it for one of the invocations. Could we move this command list into a variable and then just use that variable for both invocations?